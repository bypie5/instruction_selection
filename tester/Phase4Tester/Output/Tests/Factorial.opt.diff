--- SelfTestCases/Factorial.opt.out	2011-04-28 14:55:21.000000000 -0700
+++ ./Output/Tests/Factorial.opt.out	2020-03-13 21:47:58.000000000 -0700
@@ -1,2 +1,5 @@
-3628800
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/Factorial.opt.compiler.out line 10 column 1: "li": Too many or incorrectly formatted operands. Expected: li $t1,-100
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/Factorial.opt.compiler.out line 20 column 1: "la": Too few or incorrectly formatted operands. Expected: la $t1,($t2)
+Error in /Users/brandonyi/Documents/School/CS179E/instruction_selection/tester/Phase4Tester/./Output/Tests/Factorial.opt.compiler.out line 2 column 1: Symbol "empty_Fac" not found in symbol table.
 
+Processing terminated due to errors.
