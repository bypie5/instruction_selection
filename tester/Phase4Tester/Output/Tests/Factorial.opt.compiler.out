.data
empty_Fac

.text
jal Main
li $v0 10
syscall

Main:
li $a0 :empty_Fac
li $a1 10

move $t0 $v0
move $a0 $t0
jal _print



Fac.ComputeFac:
la $t9 $s0
sw $t9 0($sp)
move $t0 $a0
move $s0 $a1
slti $t1 $s0 1
beqz $t1 if1_else
li $t1 1
j if1_end
if1_else:
subu $t2 $s0 1
move $a0 $t0
move $a1 $t2

move $t2 $v0
mul $t1 $s0 $t2
if1_end:
move $v0 $t1
lw $s0 0($sp)




_print:
li $v0 1
syscall
la $a0 _newline
li $v0 4
syscall
jr $ra

_error:
li $v0 4
syscall
li $v0 10
syscall

_heapAlloc:
li $v0 9
syscall
jr $ra

.data
.align 0
_newline: .asciiz "\n"
_str0: .asciiz "null pointer\n"
